functional_programming_tests/test/without_functional_programming.cpp

```diff
@@ -331,32 +292,37 @@ class TaskPlanningFixture : public testing::Test {
     auto generate_path_result =
         path_generator_client_->async_send_request(request);
 
-    EXPECT_EQ(rclcpp::spin_until_future_complete(node_, generate_path_result),
-              rclcpp::FutureReturnCode::SUCCESS)
-        << "Generating path failed";
-
-    return generate_path_result.get();
+    return std::make_pair(
+        generate_path_result.get(),
+        rclcpp::spin_until_future_complete(node_, generate_path_result));
   }
 
- private:
   // Member variables
   rclcpp::Node::SharedPtr node_;
-  rclcpp::Executor::SharedPtr executor_;
-  std::thread executor_thread_;
 
   rclcpp::Client<example_srvs::srv::SetMap>::SharedPtr map_setter_client_;
   rclcpp::Client<example_srvs::srv::GetPath>::SharedPtr path_generator_client_;
-
-  std::shared_ptr<PathGenerator> pg_;
 };
 
 TEST_F(TaskPlanningFixture, same_start_and_goal) {
-  // Note all the extra boilerplate code that had to be written to
-  // test the generate_global_path function. There could have been
-  // bugs introduced in the code.
+  // Create executors process requests to the PathGenerator Server
+  auto executor = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+  std::thread executor_thread;
+
+  // The PathGenerator class to test
+  auto pg = std::make_shared<PathGenerator>();
+
+  // Starts processing requests to the PathGenerator service in a separate
+  // thread
+  executor->add_node(pg);
+  executor_thread = std::thread([&executor]() { executor->spin(); });
```

Comment:
```suggestion
  // The PathGenerator class to test
  auto const pg = std::make_shared<PathGenerator>();

  // Starts processing requests to the PathGenerator service in a separate
  // thread
  auto executor = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
  executor->add_node(pg);
  std::thread executor_thread = std::thread([&executor]() { executor->spin(); });
```
Just a bit of code golf